client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -37,16 +38,14 @@ export default React.createClass( {
 	render() {
 		let title;
 
-		if ( this.props.postViewsList.response.post && this.props.postViewsList.response.post.post_title ) {
-			title = this.translate( 'Stats for %(posttitle)s', {
-				comment: 'Title of the individual post stats page.',
-				args: {
-					posttitle: this.props.postViewsList.response.post.post_title
-				}
-			} );
-		}
+		const post = this.props.postViewsList.response.post;
+		const postOnRecord = post && post.post_title !== null;
 
-		if ( this.props.postViewsList.isError() ) {
+		if ( postOnRecord && ! this.props.postViewsList.isError() ) {
```

Comment:
I don't remember a prior bug necessitating this logic specifically. I think we were just using the convention at the time, and that's why it was there. If it still works as intended when data fails to load, then it's probably not needed